File: classes/Models/Schedule.java

import com.google.gson.annotations.SerializedName;
import com.uth.uth.R;
import com.uth.uth.api.model.clazz.enrollment.Class;

import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.AVAILABILITY;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.CAMPUS_CODE;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.CLASSROOM;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.DAYS;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.END_TIME;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.GROUP;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.IDENTIFIER;
import static com.uth.uth.api.model.clazz.Schedule.SerializedNames.START_TIME;
import static com.uth.uth.api.model.clazz.enrollment.Campus.SerializedNames.CODE;
import static com.uth.uth.util.Constants.Annotation.UNUSED_DECLARATION;

public class Schedule extends Class {

    private static final String DAYS_REGEX = "-";


    @SerializedName(AVAILABILITY)
    @SuppressWarnings(UNUSED_DECLARATION)
    private int mAvailability;

    @SerializedName(GROUP)
    @SuppressWarnings(UNUSED_DECLARATION)
    private int mGroup;


    @SerializedName(CODE)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mCampus;

    @SerializedName(CAMPUS_CODE)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mCampusCode;

    @SerializedName(CLASSROOM)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mClassroom;

    @SerializedName(DAYS)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mDays;

    @SerializedName(END_TIME)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mEndTime;

    @SerializedName(IDENTIFIER)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mIdentifier;

    @SerializedName(START_TIME)
    @SuppressWarnings(UNUSED_DECLARATION)
    private String mStartTime;


    public final int getAvailability() {
        return mAvailability;
    }

    public final int getGroup() {
        return mGroup;
    }


    @Nullable
    public final ArrayList<Day> getDays() {
        if (!StringUtils.isEmpty(mDays)) {
            final ArrayList<Day> days = new ArrayList<>();

            for (final String day : mDays.split(DAYS_REGEX)) {
                days.add(Day.get(day));
            }

            return days;
        }

        return null;
    }

    @Nullable
    public final String getCampus() {
        return mCampus;
    }

    @Nullable
    public final String getCampusCode() {
        return mCampusCode;
    }

    @Nullable
    public final String getClassroom() {
        return mClassroom;
    }

    @Nullable
    public final String getEndTime() {
        return mEndTime;
    }

    @Nullable
    public final String getIdentifier() {
        return mIdentifier;
    }

    @Nullable
    public final String getStartTime() {
        return mStartTime;
    }


    public enum Day {

        SUNDAY("DO", R.string.title_weekday_sunday),

        MONDAY("LU", R.string.title_weekday_monday),

        TUESDAY("MA", R.string.title_weekday_tuesday),

        WEDNESDAY("MI", R.string.title_weekday_wednesday),

        THURSDAY("JU", R.string.title_weekday_thursday),

        FRIDAY("VI", R.string.title_weekday_friday),

        SATURDAY("SA", R.string.title_weekday_saturday);

        private static final Map<String, Day> mMap;

        static {
            mMap = new HashMap<>();

            for (final Day day : Day.values()) {
                mMap.put(day.mDay, day);
            }
        }

        @StringRes
        private final int mStringResId;

        private final String mDay;

        Day(@NonNull final String day, @StringRes final int stringRes) {
            mDay = day;

            mStringResId = stringRes;
        }

        @Nullable
        private static Day get(@NonNull final String day) {
            return mMap.get(day);
        }

        @StringRes
        public final int getStringResId() {
            return mStringResId;
        }

    }


    public static class SerializedNames {

        public static final String GROUP = "group";

        public static final String IDENTIFIER = "identifier";


        static final String AVAILABILITY = "availability";

        static final String CAMPUS_CODE = "campus_code";

        static final String CLASSROOM = "classroom";

        static final String DAYS = "days";

        static final String END_TIME = "end_time";

        static final String START_TIME = "start_time";

    }

}

File: classes/Models/Schedule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Project.Models
{
    public class Schedule: BindableBase
    {

        private string _campusCode;

        public string CampusCode
        {
            get { return _campusCode; }
            set { SetProperty(ref _campusCode, value); }
        }


        private string _campus;

        public string Campus
        {
            get { return _campus; }
            set { SetProperty(ref _campus, value); }
        }


        private string _name;

        public string Name
        {
            get { return _name; }
            set { SetProperty(ref _name, value); }
        }


        private string _days;

        public string Days
        {
            get { return _days; }
            set { SetProperty(ref _days, value); }
        }


        private string _classroom;

        public string Classroom
        {
            get { return _classroom; }
            set { SetProperty(ref _classroom, value); }
        }


        private string _startTime;

        public string StartTime
        {
            get { return _startTime; }
            set { SetProperty(ref _startTime, value); }
        }


        private string _endTime;

        public string EndTime
        {
            get { return _endTime; }
            set { SetProperty(ref _endTime, value); }
        }


        private string _code;

        public string Code
        {
            get { return _code; }
            set { SetProperty(ref _code, value); }
        }


        private string _group;

        public string Group
        {
            get { return _group; }
            set { SetProperty(ref _group, value); }
        }


        private string  _courseInitial;

        public string  CourseInitial
        {
            get { return _courseInitial; }
            set { _courseInitial = value; }
        }


        private string _colorListItem;

        public string ColorListItem
        {
            get { return _colorListItem; }
            set { _colorListItem = value; }
        }
        
        

        


    }
}
